include/tvm/relay/pass.h

```diff
@@ -144,59 +144,65 @@ tvm::Array<Var> AllVars(const Expr& expr);
  * type in the context.
  *
  * \param expr the expression.
+ * \param mod the module.
  *
  * \return List of free vars, in the PostDFS order visited by expr.
  */
-tvm::Array<TypeVar> FreeTypeVars(const Expr& expr);
+tvm::Array<TypeVar> FreeTypeVars(const Expr& expr, const Module& mod);
```

Comment:
Why do we now need the module on all these functions?

Reply:
For ADTs, we need to know if there are type vars in the definitions

Reply:
Why is that? can you illustrate an example for me?

Reply:
A constructor can contain type vars that are bound only in the type data. If someone uses the wrong type var in a constructor (i.e., one that is not bound in the type data), the FreeVars pass can catch that.

Might be a pain to have to pass the module for just this marginal case, though.